File: src/types/address.ts
export type AddressLabel =
  | 'Home'
  | 'Office'
  | 'Other';

export interface Address {
  id: string;

  fullName: string;

  phoneNumber: string;

  house: string;

  area: string;

  landmark?: string;

  city: string;

  pinCode: string;

  label: AddressLabel;

  isDefault: boolean;
}

File: backend/app/schemas/address.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class AddressCreate(BaseModel):
    user_id: str
    label: str
    full_name: str
    phone: str
    address_line: str
    landmark: Optional[str] = None
    city: str
    state: str
    pincode: str
    is_default: bool = False


class AddressUpdate(BaseModel):
    label: Optional[str] = None
    full_name: Optional[str] = None
    phone: Optional[str] = None
    address_line: Optional[str] = None
    landmark: Optional[str] = None
    city: Optional[str] = None
    state: Optional[str] = None
    pincode: Optional[str] = None
    is_default: Optional[bool] = None


class AddressResponse(AddressCreate):
    id: str
    created_at: datetime
    updated_at: datetime